evalml/data_checks/invalid_targets_data_check.py

```diff
@@ -80,14 +80,14 @@ def validate(self, X, y):
             return results
 
         y = infer_feature_types(y)
-        is_supported_type = y.ww.logical_type in numeric_and_boolean_ww + [
+        is_supported_type = type(y.ww.logical_type) in numeric_and_boolean_ww + [
```

Comment:
I think this can be `isinstance(y.ww.logical_type, tuple(numeric_and_boolean_ww))` if you'd like here and in the other places where a similar check.

Reply:
@tamargrey I think the other option is to use `y.ww.logical_type.type_string` and then update our lists, equality/member checks to use the `type_strings`. What are your thoughts? I think both do the same thing but using `type_string` may be backwards compatible?

Reply:
@freddyaboulton I like that idea! It does seem like it would be backwards compatible, which is nice.

It's definitely different from how we've tended to compare logical types inside of Woodwork, so my one worry is that it's using the `type_string` for something that it wasn't intended for. I'm not sure that that's actually a problem, though.